mcp/communications/tests/test_evtm.c

```diff
@@ -0,0 +1,385 @@
+ *
+ * @brief This file is part of MCP, created for the Terahertz Intensity Mapper (TIM) project.
+ *
+ * This software is copyright (C) 2023 University of Pennsylvania.
+ *
+ * MCP is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * MCP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MCP; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <cmocka.h>
+#include <float.h>
+#include <stdlib.h>
+#include <mcp.h>
+
+#include "evtm.c"
+#include "mcp_mock_decl.c"
+#include "blast.h"
+#include "channels_tng.h"
+#include "FIFO.h"
+
+// if compiler is invoked with --wrap=<func>, then the linker will resolve
+// <func> to __wrap_<func> instead of <func>.
+// this allows us to mock BITServer behavior.
+
+// function_called allows us to check how many times that function was called
+// unfortunately, it does not let us check if that count is 0 :(
+
+void __wrap_bprintf(buos_t l, const char *fmt, ...) {
+    char message[BUOS_MAX];
+    va_list argptr;
+
+    va_start(argptr, fmt);
+    vsnprintf(message, BUOS_MAX, fmt, argptr);
+    va_end(argptr);
+
+    bputs(info, message);
+
+    if (l == fatal) {
+        check_expected(fmt);
+        function_called();
+    }
+}
+
+int __wrap_initBITSender(struct BITSender *server, const char *send_addr,
+  unsigned int port, unsigned int fifo_length, unsigned int fifo_maxsize, unsigned int packet_maxsize) {
+    assert_non_null(server);
+    assert_int_equal(fifo_length, FIFO_LEN);
+    assert_int_equal(fifo_maxsize, MAX(EVTM_MAX_SIZE, superframe->allframe_size*2));
+    assert_int_equal(packet_maxsize, EVTM_MAX_PACKET_SIZE);
+
+    check_expected(send_addr);
+    check_expected(port);
+    function_called();
+    return mock_type(int);
+}
+
+void __wrap_setBITSenderSerial(struct BITSender *sender, uint32_t serial) {
+    function_called();
+    assert_non_null(sender);
+    check_expected(serial);
+}
+
+int __wrap_setBITSenderFramenum(struct BITSender *sender, uint32_t framenum) {
+    function_called();
+    assert_non_null(sender);
+    check_expected(framenum);
+    return mock_type(int);
+}
+
+int __wrap_sendToBITSender(struct BITSender *sender, uint8_t *data, unsigned int size, uint8_t priority) {
+    function_called();
+    assert_non_null(sender);
+    check_expected(data);
+    check_expected(size);
+    check_expected(priority);
+    return mock_type(int);
+}
+
+int __wrap_testing_evtm() {
```

Comment:
I suggested updating this function name, so it would have to be updated here as well